docs/third-party-developers/extensibility/data-store/payment.md

```diff
@@ -444,6 +446,52 @@ const store = select( 'wc/store/payment' );
 const paymentResult = store.getPaymentResult();
 ```
 
+### getIncompatiblePaymentMethods
+
+Returns the list of payment methods that are incompatible with Checkout block.
+
+#### _Returns_ <!-- omit in toc -->
+
+-   `object`: A list of incompatible payment methods with the following properties:
+   	-  _name_ `string`: The name of the payment method.
+
+#### _Example_ <!-- omit in toc -->
+
+```js
+const store = select( 'wc/store/payment' );
+const incompatiblePaymentMethods = store.getIncompatiblePaymentMethods();
+```
+
+### getState
+
+Returns the current state of the payment store.
```

Comment:
What are your thoughts on adding the warning from the original function to ensure that this selector is not accidentally used?

```suggestion
Returns the current state of the payment store.

> 🚨 Instead of using this selector, the focused selectors should be used. This selector should only be used to mock selectors in our unit tests.
```

Here's the original function:
```ts
// We should avoid using this selector and instead use the focused selectors
// We're keeping it because it's used in our unit test: assets/js/blocks/cart-checkout-shared/payment-methods/test/payment-methods.js
// to mock the selectors.
export const getState = ( state: PaymentState ) => {
	return state;
};
```